include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -184,72 +195,79 @@ struct CollectiveMma<
-        make_layout(make_shape(_1{}, _1{}, K),
-                    make_stride(_1{}, E<0>{}, E<1>{})));
-    Tensor tBi = make_tensor(
-        make_inttuple_iter(
-            *gB.data() +
-            make_coord((sub_group_id / sg_per_wg_n % 2 * 2) * get<1>(MmaAtomShape{}),
-                       (sub_group_id / sg_per_wg_n / 2 % 2) * get<2>(MmaAtomShape{}))),
-        make_layout(make_shape(_1{}, _1{}, K),
-                    make_stride(_1{}, E<0>{}, E<1>{})));
     //
     // Mainloop
     //
-    int prefetch_k = 0;
-
+    const int m_coord = BlockIdxX() * BLK_M + (get_sub_group_id() / ATOM_N) * SG_M;
+    const int n_coord = BlockIdxY() * BLK_N + (get_sub_group_id() % ATOM_N) * SG_N;
+    const int l_coord = BlockIdxZ();
+    Tensor iter_a = mainloop.gmem_tiled_copy_a.get_pvc_tensor(
+      make_coord(m_coord, 0, l_coord), append<4>(tCrA_copy_view.shape(), k_tile_count),
+      append<3>(typename XE_Copy_A::Shape_MN{}, BLK_K), seq<0,1,1>{});
+    Tensor iter_b = mainloop.gmem_tiled_copy_b.get_pvc_tensor(
+      make_coord(0, n_coord, l_coord), append<4>(tCrB_copy_view.shape(), k_tile_count),
+      append<3>(typename XE_Copy_B::Shape_MN{}, BLK_K), seq<0,1,0>{});
+#pragma unroll
     for (int i = 0; i < DispatchPolicy::Stages; i++) {
-      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
-      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, _, prefetch_k));
-      prefetch_k += get<2>(SubgroupTileShape{});
+      if constexpr(cute::detail::has_prefetch<GmemTiledCopyA>) {
+        prefetch(mainloop.gmem_tiled_copy_a, iter_a(_,_,_,i));
+      }
+      if constexpr(cute::detail::has_prefetch<GmemTiledCopyB>) {
+        prefetch(mainloop.gmem_tiled_copy_b, iter_b(_,_,_,i));  
+      }
     }
-
-    for (int k_tile = 0, k = 0; k_tile < k_tile_count;
-         ++k_tile, k += get<2>(SubgroupTileShape{})) {
+#pragma unroll
+    for (int k_tile = 0; k_tile < k_tile_count; ++k_tile) {
       // Copy gmem to rmem for the first k_tile
-      copy(mainloop.gmem_tiled_copy_a, gA(_, _, k), tAr);
-      copy(mainloop.gmem_tiled_copy_b, gB(_, _, k), tBr);
-
-      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
-      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, _, prefetch_k));
-      prefetch_k += get<2>(SubgroupTileShape{});
-
-      cute::gemm(tiled_mma, accum, tAr_view, tBr_view, src_accum);
+      copy(mainloop.gmem_tiled_copy_a, iter_a(_,_,_,k_tile), tCrA_copy_view);
+      copy(mainloop.gmem_tiled_copy_b, iter_b(_,_,_,k_tile), tCrB_copy_view);
+    if(k_tile + DispatchPolicy::Stages < k_tile_count) {
+      if constexpr(cute::detail::has_prefetch<GmemTiledCopyA>) {
+        prefetch(mainloop.gmem_tiled_copy_a, iter_a(_,_,_,k_tile + DispatchPolicy::Stages));
+      }
+      if constexpr(cute::detail::has_prefetch<GmemTiledCopyB>) {
+        prefetch(mainloop.gmem_tiled_copy_b, iter_b(_,_,_,k_tile + DispatchPolicy::Stages));  
+      }
+    }
+    for (int i = 0; i < SG_K / SubgroupSize; i++) {
```

Comment:
This loop shouldn't be needed if you use tCrB_partition

Reply:
Due to the above reason, we have to unloop this